pkg/executor/sortexec/sort.go

```diff
@@ -52,36 +56,61 @@ type SortExec struct {
 	memTracker  *memory.Tracker
 	diskTracker *disk.Tracker
 
-	// sortPartitions is the chunks to store row values for partitions. Every partition is a sorted list.
-	sortPartitions []*sortPartition
+	IsUnparallel bool
+
+	Unparallel struct {
+		Idx int
+
+		// sortPartitions is the chunks to store row values for partitions. Every partition is a sorted list.
+		sortPartitions []*sortPartition
+
+		// multiWayMerge uses multi-way merge for spill disk.
+		// The multi-way merge algorithm can refer to https://en.wikipedia.org/wiki/K-way_merge_algorithm
+		multiWayMerge *multiWayMerge
+
+		// spillAction save the Action for spill disk.
+		spillAction *sortPartitionSpillDiskAction
+	}
+
+	Parallel struct {
+		result sortedRows
+		rowNum int64
+		idx    int64
 
-	// multiWayMerge uses multi-way merge for spill disk.
-	// The multi-way merge algorithm can refer to https://en.wikipedia.org/wiki/K-way_merge_algorithm
-	multiWayMerge *multiWayMerge
-	// spillAction save the Action for spill disk.
-	spillAction *sortPartitionSpillDiskAction
+		chunkChannel    chan *chunkWithMemoryUsage
+		isChannelClosed *atomic.Bool
+		workers         []*parallelSortWorker
+
+		// All workers' sorted rows will be put into this list to be merged
+		globalSortedRowsQueue *sortedRowsList
+
+		errRWLock *sync.RWMutex
+		err       error
+	}
 
 	enableTmpStorageOnOOM bool
 }
 
 // Close implements the Executor Close interface.
 func (e *SortExec) Close() error {
-	for _, partition := range e.sortPartitions {
-		err := partition.close()
-		if err != nil {
-			return err
+	if e.Unparallel.spillAction != nil {
+		e.Unparallel.spillAction.SetFinished()
+	}
+
+	if e.IsUnparallel {
+		for _, partition := range e.Unparallel.sortPartitions {
+			partition.close()
+		}
+	} else {
+		if e.Parallel.err != nil {
```

Comment:
Need to ensure that all goroutines have exited.
And always close children(0).

Reply:
> Need to ensure that all goroutines have exited. And always close children(0).

done